submissions/Zakkarat/JS-OOP/js/app.js

```diff
@@ -0,0 +1,142 @@
+const startX = 200;
+const startY = 380;
+const stepX = 100;
+const stepY = 80;
+const scr = document.querySelectorAll('p');
+const mdl = document.getElementsByClassName('modal')[0];
+const mdlCont = document.getElementsByClassName('modal-content')[0];
+const btn = document.getElementsByTagName('button')[0];
+let allowMove = true;
+
+var Enemy = function(y) {
+    this.speed = speedRand();
+    this.x = positionRand();
+    this.y = y;
+    this.sprite = 'images/enemy-bug.png';
+};
+
+function speedRand() {
+  let speed = Math.floor(Math.random() * (375 - 200 ) + 200);
+  return speed;
+}
+function positionRand() {
+  let position = Math.floor(Math.random() * (-225))
+  return position;
+}
+
+Enemy.prototype.colision = function() {
+  return (Math.abs(this.x - player.x) < 60 && Math.abs(this.y - player.y) < 70)
+}
+
+Enemy.prototype.update = function(dt) {
+    this.x += this.speed * dt;
+    if (this.x > 500) {
+      this.x = positionRand();
+      this.speed = speedRand();
+    }
+    if (this.colision()) {
+      player.x = startX;
+      player.y = startY;
+      updLive();
+    }
+};
+
+Enemy.prototype.render = function() {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+const Player = function(x, y) {
+  this.x = x;
+  this.y = y;
+  this.sprite = 'images/char-boy.png';
+}
+Player.prototype.render = function() {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+Player.prototype.update = function() {
+    if(this.y < 0){
+      this.x = startX;
+      this.y = startY;
+      updScore();
+    }
+  }
+
+Player.prototype.handleInput = function(allowedKey) {
+    switch (allowedKey){
+        case 'left':
+          if(this.x > 0 && allowMove) {
+            this.x -= stepX;
+          }
+            break;
+        case 'up':
+          if(allowMove) {
+            this.y -= stepY;
+          }
+            break;
+        case 'right':
+          if(this.x < 400 && allowMove) {
+            this.x += stepX;
+          }
+            break;
+        case 'down':
+          if(this.y < 370 && allowMove) {
+            this.y += stepY;
+          }
+            break;
+    }
+};
+// Now instantiate your objects.
+// Place all enemy objects in an array called allEnemies
+// Place the player object in a variable called player
+let player = new Player(startX, startY);
+
+let allEnemies = [new Enemy(60), new Enemy(140), new Enemy(230)];
```

Comment:
* [ ] arrays of constants are also constants
         (e.g. `const INITIAL_POSITIONS = [1,2,3,4].map(rowNumber => rowNumber * BLOCK_HEIGHT);`)